mmedit/models/backbones/sr_backbones/glean_styleganv2.py

```diff
@@ -0,0 +1,315 @@
+        style_channels (int): The number of channels for style code.
+        num_mlps (int, optional): The number of MLP layers. Defaults to 8.
+        channel_multiplier (int, optional): The mulitiplier factor for the
+            channel number. Defaults to 2.
+        blur_kernel (list, optional): The blurry kernel. Defaults
+            to [1, 3, 3, 1].
+        lr_mlp (float, optional): The learning rate for the style mapping
+            layer. Defaults to 0.01.
+        default_style_mode (str, optional): The default mode of style mixing.
+            In training, we defaultly adopt mixing style mode. However, in the
+            evaluation, we use 'single' style mode. `['mix', 'single']` are
+            currently supported. Defaults to 'mix'.
+        eval_style_mode (str, optional): The evaluation mode of style mixing.
+            Defaults to 'single'.
+        mix_prob (float, optional): Mixing probabilty. The value should be
+            in range of [0, 1]. Defaults to 0.9.
+        pretrained (dict | None, optional): Information for pretained models.
+            The necessary key is 'ckpt_path'. Besides, you can also provide
+            'prefix' to load the generator part from the whole state dict.
+            Defaults to None.
+        bgr2rgb (bool, optional): Whether to flip the image channel dimension.
+            Defaults to False.
+    """
+
+    def __init__(self,
+                 in_size,
+                 out_size,
+                 style_channels,
+                 num_mlps=8,
+                 channel_multiplier=2,
+                 blur_kernel=[1, 3, 3, 1],
+                 lr_mlp=0.01,
+                 default_style_mode='mix',
+                 eval_style_mode='single',
+                 mix_prob=0.9,
+                 pretrained=None,
+                 bgr2rgb=False):
+
+        super().__init__()
+
+        # latent bank (StyleGANv2), with weights being fixed
+        self.generator = build_component(
+            dict(
+                type='StyleGANv2Generator',
+                out_size=out_size,
+                style_channels=style_channels,
+                num_mlps=num_mlps,
+                channel_multiplier=channel_multiplier,
+                blur_kernel=blur_kernel,
+                lr_mlp=lr_mlp,
+                default_style_mode=default_style_mode,
+                eval_style_mode=eval_style_mode,
+                mix_prob=mix_prob,
+                pretrained=pretrained,
+                bgr2rgb=bgr2rgb))
+        self.generator.requires_grad_(False)
+
+        self.in_size = in_size
+        self.style_channels = style_channels
+        channels = self.generator.channels
+
+        # encoder
+        num_styles = int(math.log2(out_size)) * 2 - 2
+        encoder_res = [2**i for i in range(int(math.log2(in_size)), 1, -1)]
+        self.encoder = nn.ModuleList()
+        self.encoder.append(
+            nn.Sequential(
+                RRDBFeatureExtractor(3, 64, num_blocks=23),
```

Comment:
Maybe adding `mid_channels`(defaults 64) , `in_channels` and `out_channels` to the class (GLEANStyleGANv2) will be better.